### ISSUE TITLE
box.auth should throw an exception on errors

### ISSUE THREAD

**Issue by dkent600** [3box/3box-js#872]
After this:

```
    box = await Box.create(web3Provider);
    await box.auth([spaceName], { address: accountAddress });
```

I see this in the console:

> Saving profile to 3box failed: Could not append entry, key "did:3:bafyreictkuq3l27eklrfdathwhlynfftbtjuhkyjdavojnwa5qp52jhdqi" is not allowed to write to the log

Besides that I don't know what that message means, `box.auth` does not throw an exception, which is a problem for any dApp wanting to properly handle errors.

Using 3box@1.20.3

**Comment by oed** [comment]
The problem here is that the error comes from OrbitDB and is thrown in some background process, so it's not something we can easily catch. 

@zachferland has talked about that there might be some fix to be made in Orbitdb, but I'm not sure of the status of that.

**Comment by aturX** [comment]
"Error: Could not append entry, key "did:3:bafyreig4trqtifolb3mw7b6elt5butfb23gj2n6dugtjm2gwt63s..." is not allowed to write to the log"

I have same problem , I can read data from 3box, but can't change or add new data into.

this is my code 
`   
this.box = await Box.openBox(this.address, window.ethereum)
            this.box.syncDone

            console.log(this.box)

            // this.space = await this.box.openSpace('ant-cave')
            // await this.space.syncDone

            const spaces = ['ant-cave']
            const address = this.address
            await this.box.auth(spaces, { address })`

**Comment by aphelionz** [comment]
Looks like the OrbitDB udpate solved the problem for @aturX :+1:

**Comment by oed** [comment]
Will be fixed by #901 
cc @zachferland

**Comment by zachferland** [comment]
3box@1.22.0 has been release now and includes orbit-db@0.25.1 which should include a fix for this error

Closing issue now, please re open another if any issues continue!
